projects/core/src/events/event.register.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable, Type } from '@angular/core';
+import { merge, Observable, of } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class EventRegister {
+  private events = {};
```

Comment:
What about using Map instead of plain object here, you can use an actual class as a key (also eventClass.name won't probably work in AOT).